cmd/osv-scanner/fix/noninteractive.go

```diff
@@ -295,6 +295,19 @@ func autoOverride(ctx context.Context, r reporter.Reporter, opts osvFixOptions,
 		return err
 	}
 
+	if opts.ManifestRW.System() == resolve.Maven {
+		// Update Maven registries based on the pom.xml
+		specific, ok := manif.EcosystemSpecific.(manifest.MavenManifestSpecific)
```

Comment:
are there other places where we should be checking for the registries to use beyond the pom.xml? i.e. per https://maven.apache.org/guides/mini/guide-multiple-repositories.html#repository-order

```
effective settings:
Global settings.xml
User settings.xml
local effective build POM:
Local pom.xml
Parent POMs, recursively
Super POM
effective POMs from dependency path to the artifact.
```

We need to check `settings.xml` in various places, as well as the parent POMs? Do we need to add a TODO here?

Reply:
repositories defined in parent POMs will be merged when we merge other data from parents.

I made an [issue](https://github.com/google/osv-scanner/issues/1269) for reading settings.xml, but I think it's a good idea to refer to that issue here.